### PR TITLE
feat: new extension `vike-solid-query`

### LINKED ISSUE #94: New extension `vike-solid-query`?
See for example:
 - https://github.com/vikejs/vike-react/tree/main/packages/vike-react-query
 - https://github.com/vikejs/vike-vue/tree/main/packages/vike-vue-query

### PR DESCRIPTION
**Pull request by phonzammi**
@magne4000 @brillout 

Please take a look and review this PR.

Note that I haven’t added any tests for `vike-solid-query` yet because we need to add some code to `vike-solid` to detect `crawlers and bots`, and then use [renderToStringAsync()](https://docs.solidjs.com/reference/rendering/render-to-string-async#rendertostringasync) for that. Also a bit of refactor.

As Rom suggested, maybe we could merge this since we've refactored to a monorepo and continue with the remaining work in a separate PR.

closes #94

### CODE REVIEW

File: packages/vike-solid-query/README.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+[See also](#see-also)  
+
+## Installation
+
+1. `npm install @tanstack/solid-query vike-solid-query`
+2. Extend `+config.js`:
+   ```js
+   // pages/+config.js
+
+   import vikeSolid from 'vike-Solid/config'
```
Comment (magne4000) [inline packages/vike-solid-query/README.md:26]:
```suggestion
   import vikeSolid from 'vike-solid/config'
```

File: packages/vike-solid-query/README.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+function Movies() {
+  const query = useSuspenseQuery({
+    queryKey: ['movies'],
+    queryFn: () => fetch('https://star-wars.brillout.com/api/films.json')
+  })
+  const query = createQuery(() => ({
+    queryKey: ["movies"],
+    queryFn: () => fetch('https://star-wars.brillout.com/api/films.json')
+  }));
```
Comment (magne4000) [inline packages/vike-solid-query/README.md:140]: The first one should probably be removed?

**Review by magne4000 — COMMENTED** [review]
Just a few typos to fix. It also needs to be updated from `main` branch. After that, we are good to merge!

**Comment by phonzammi** [comment]
@magne4000, LGTM ?

@brillout, we’d love to hear your thoughts on this.

**Comment by magne4000** [comment]
> As Rom suggested, maybe we could merge this since we've refactored to a monorepo and continue with the remaining work in a separate PR.

Let's just merge and continue the work on another PR :+1:

**Review by magne4000 — APPROVED** [review]

**Comment by brillout** [comment]
I quickly glazed over the readme; looks good. I wonder if it's possible to have the interface be more like `vike-react-query`'s `withFallback()`, but I guess it's good as it is.

FYI there are couple of typos, inconsistent code formatting, and markdown formatting errors in the readme.

Nothing much changed on `vike-solid'`s side, so there isn't much for me to object.

LGTM.

**Comment by brillout** [comment]
Also, the main readme needs to be updated.

I've aded `vike-solid-query` to Vike's docs: https://github.com/vikejs/vike/commit/9711fbb0549857433270b6e19b77a4b333faccc3 :rocket:

**Pull request merged**